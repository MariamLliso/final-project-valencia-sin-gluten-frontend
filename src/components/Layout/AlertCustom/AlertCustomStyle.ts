import styled from "styled-components";


const AlertCustomStyle = styled.div`
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background: rgba( 255, 255, 255, 0.8);
  padding: 1.5rem clamp(1rem, 5%, 3rem)
`;

export default AlertCustomStyle;
