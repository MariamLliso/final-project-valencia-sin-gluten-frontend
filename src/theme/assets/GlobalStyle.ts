import { createGlobalStyle } from "styled-components";

const GlobalStyle = createGlobalStyle`
  h1 {
    margin:0;
  }

  ul {
    list-style: none;
    margin: 0;
    padding-left: 0;
  }
`;

export default GlobalStyle;
