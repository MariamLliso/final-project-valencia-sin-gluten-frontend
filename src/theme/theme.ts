import { alpha, createTheme } from "@mui/material";

export const pxToRem = (value: number) => {
  return `${value / 16}rem`;
};

interface IMediaSizes {
  sm: number;
  md: number;
  lg: number;
}

export const responsiveFontSizes = ({ sm, md, lg }: IMediaSizes) => {
  return {
    "@media (min-width:600px)": {
      fontSize: pxToRem(sm),
    },
    "@media (min-width:900px)": {
      fontSize: pxToRem(md),
    },
    "@media (min-width:1200px)": {
      fontSize: pxToRem(lg),
    },
  };
};

export const fontPrimary = "Poppins, sans-serif";
export const fontSecondary = "Libre Bodoni, serif;";

const greyColor = {
  0: "#FFFFFF",
  100: "#F5F5F5",
  200: "#F4F6F8",
  300: "#DFE3E8",
  400: "#C4CDD5",
  500: "#919EAB",
  600: "#637381",
  700: "#454F5B",
  800: "#212B36",
  900: "#161C24",
  500_8: alpha("#919EAB", 0.08),
  500_12: alpha("#919EAB", 0.12),
  500_16: alpha("#919EAB", 0.16),
  500_24: alpha("#919EAB", 0.24),
  500_32: alpha("#919EAB", 0.32),
  500_48: alpha("#919EAB", 0.48),
  500_56: alpha("#919EAB", 0.56),
  500_80: alpha("#919EAB", 0.8),
};

const primaryColor = {
  light: "#E9F3D9",
  main: "#5FAA46",
  dark: "#3b8224",
  contrastText: "#fff",
};

const secondaryColor = {
  lighter: "#D6E4FF",
  light: "#84A9FF",
  main: "#3366FF",
  dark: "#1939B7",
  darker: "#091A7A",
  contrastText: "#fff",
};

const theme = createTheme({
  typography: {
    fontFamily: fontPrimary,
    fontWeightRegular: 400,
    fontWeightMedium: 500,
    fontWeightBold: 600,
    h1: {
      fontFamily: fontSecondary,
      fontWeight: 700,
      lineHeight: 80 / 64,
      fontSize: pxToRem(40),
      ...responsiveFontSizes({ sm: 52, md: 58, lg: 64 }),
    },
    h2: {
      fontFamily: fontSecondary,
      fontWeight: 700,
      lineHeight: 64 / 48,
      fontSize: pxToRem(32),
      ...responsiveFontSizes({ sm: 40, md: 44, lg: 48 }),
    },
    h3: {
      fontFamily: fontSecondary,
      fontWeight: 700,
      lineHeight: 1.5,
      fontSize: pxToRem(24),
      ...responsiveFontSizes({ sm: 26, md: 30, lg: 32 }),
    },
    h4: {
      fontWeight: 700,
      lineHeight: 1.5,
      fontSize: pxToRem(20),
      ...responsiveFontSizes({ sm: 20, md: 24, lg: 24 }),
    },
    h5: {
      fontWeight: 700,
      lineHeight: 1.5,
      fontSize: pxToRem(18),
      ...responsiveFontSizes({ sm: 19, md: 20, lg: 20 }),
    },
    h6: {
      fontWeight: 700,
      lineHeight: 28 / 18,
      fontSize: pxToRem(17),
      ...responsiveFontSizes({ sm: 18, md: 18, lg: 18 }),
    },
    subtitle1: {
      fontWeight: 600,
      lineHeight: 1.5,
      fontSize: pxToRem(16),
    },
    subtitle2: {
      fontWeight: 600,
      lineHeight: 22 / 14,
      fontSize: pxToRem(14),
    },
    body1: {
      lineHeight: 1.5,
      fontSize: pxToRem(16),
    },
    body2: {
      lineHeight: 22 / 14,
      fontSize: pxToRem(14),
    },
    caption: {
      lineHeight: 1.5,
      fontSize: pxToRem(12),
    },
    overline: {
      fontWeight: 700,
      lineHeight: 1.5,
      fontSize: pxToRem(12),
      letterSpacing: 1.1,
      textTransform: "uppercase",
    },
    button: {
      fontWeight: 700,
      lineHeight: 24 / 14,
      fontSize: pxToRem(14),
      textTransform: "uppercase",
    },
  },
  palette: {
    common: { black: "#000", white: "#fff" },
    primary: { ...primaryColor },
    secondary: { ...secondaryColor },
    grey: greyColor,
    text: {
      primary: greyColor[800],
      secondary: greyColor[600],
      disabled: greyColor[500],
    },
    background: { paper: "#fff", default: greyColor[0] },
    action: {
      active: greyColor[600],
      hover: greyColor[500_8],
      selected: greyColor[500_16],
      disabled: greyColor[500_80],
      disabledBackground: greyColor[500_24],
      focus: greyColor[500_24],
      hoverOpacity: 0.08,
      disabledOpacity: 0.48,
    },
  },
  components: {
    MuiButtonBase: {
      styleOverrides: {
        root: {
          letterSpacing: "0.1em",
        },
      },
      defaultProps: {
        disableRipple: true,
      },
    },
    MuiInputBase: {
      styleOverrides: {
        input: {
          "&::placeholder": {
            fontSize: "0.875rem",
          },
        },
      },
    },
    MuiOutlinedInput: {
      styleOverrides: {
        root: {
          borderRadius: `10px`,
          "&.MuiInputBase-multiline": {
            padding: 1,
          },
        },
        input: {
          fontWeight: 500,
          padding: "15.5px 14px",
          borderRadius: `10px`,
          "&.MuiInputBase-inputSizeSmall": {
            padding: "10px 14px",
            "&.MuiInputBase-inputAdornedStart": {
              paddingLeft: 0,
            },
          },
        },
        inputAdornedStart: {
          paddingLeft: 4,
        },
        notchedOutline: {
          borderRadius: `10px`,
        },
      },
    },
  },
});

export default theme;
