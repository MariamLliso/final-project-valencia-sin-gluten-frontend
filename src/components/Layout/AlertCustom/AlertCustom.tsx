import { Alert, AlertTitle } from "@mui/material";
import { ITypeOfAlert } from "../../../types/uiInterfaces";
import AlertCustomStyle from "./AlertCustomStyle";

export interface IAlertCustom {
  title: string,
  content: string,
  type: ITypeOfAlert,
  action: Function
}

const AlertCustom = ({title, content, type, action}: IAlertCustom): JSX.Element => {
  const onClose = () => {
    action();
  }

  return (
      <AlertCustomStyle>
        <Alert severity={type} onClose={onClose}>
          <AlertTitle>{title}</AlertTitle>
          {content}
        </Alert>
      </AlertCustomStyle>
  )
}

export default AlertCustom;
