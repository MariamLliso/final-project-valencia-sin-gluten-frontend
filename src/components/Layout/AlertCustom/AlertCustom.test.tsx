import { CssBaseline, ThemeProvider } from "@mui/material";
import GlobalStyle from "../../../theme/assets/GlobalStyle";
import theme from "../../../theme/theme";
import AlertCustom, { IAlertCustom } from "./AlertCustom";
import TestRenderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Given a AlerCustom component", () => {
  const givenAlertProp: IAlertCustom = {
    title: "some title",
    content: "som content",
    type: "warning",
    action: jest.fn(),
  };

  describe("When it's invoked", () => {
    test("Then it should always match this snapshot", () => {
      const testedAlertCustom = TestRenderer.create(
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <GlobalStyle />
            <CssBaseline />
            <AlertCustom
              title={givenAlertProp.title}
              content={givenAlertProp.content}
              type={givenAlertProp.type}
              action={givenAlertProp.action}
            />
          </BrowserRouter>
        </ThemeProvider>
      );

      expect(testedAlertCustom).toMatchSnapshot();
    });
  });

  describe("When it's invoked and user click at CloseIcon", () => {
    test("Then it should call the given action fuction", () => {
      render(
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <GlobalStyle />
            <CssBaseline />
            <AlertCustom
              title={givenAlertProp.title}
              content={givenAlertProp.content}
              type={givenAlertProp.type}
              action={givenAlertProp.action}
            />
          </BrowserRouter>
        </ThemeProvider>
      );

      const expectedColseButton = screen.getByTestId("CloseIcon");
      userEvent.click(expectedColseButton);

      expect(givenAlertProp.action).toHaveBeenCalled();
    });
  });
});
