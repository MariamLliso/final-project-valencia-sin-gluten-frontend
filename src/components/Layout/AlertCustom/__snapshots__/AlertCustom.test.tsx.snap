// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a AlerCustom component When it's invoked Then it should always match this snapshot 1`] = `
<div
  className="sc-bczRLJ jILTDa"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-1fi1xst-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="ReportProblemOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-acap47-MuiAlert-message"
    >
      <div
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-zahiqg-MuiTypography-root-MuiAlertTitle-root"
      >
        some title
      </div>
      som content
    </div>
    <div
      className="MuiAlert-action css-ki1hdl-MuiAlert-action"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1v8djt2-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Close"
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
